### PR TITLE
[T5A1][F11-B2]Teo Ming Yi

### PR DESCRIPTION
**Pull request by myteo**
Ready for review.

### CODE REVIEW

File: doc/DeveloperGuide.md [hunk 1]
```diff
@@ -47,6 +47,8 @@ Priority | As a ... | I want to ... | So that I can...
 `* * *` | user | find a person by name | locate details of persons without having to go through the entire list
 `* *` | user | hide [private contact details](#private-contact-detail) by default | to minimize chance of someone else seeing them by accident
 `*` | user with many persons in the address book | sort persons by name | locate a person easily
+`*` | user | find a person by phone number | locate details of persons when only given a phone number
+`*` | user | find a person by tag | find a list of persons with a specific tag
```
Comment (louietyj) [inline doc/DeveloperGuide.md]: Great work!

**Comment by myteo** [comment]
Reviewed in tutorial.

File: doc/DeveloperGuide.md [hunk 2]
```diff
@@ -74,12 +76,40 @@ Use case ends.
...
   Use case resumes at step 2
 
+#### Use case: Edit tag
+
+**MSS**
+
+1. User requests to list persons
+2. AddressBook shows a list of persons
+3. User requests to edit tag of a specific person in the list
+4. AddressBook changes the tag of the person <br>
```
Comment (JoelT-92) [inline doc/DeveloperGuide.md]: Good job. However, you can still add more interactions with the system. 
For example, you can add that the user edits the tag, system requests confirmation, and user confirms etc.

File: doc/DeveloperGuide.md [hunk 3]
```diff
@@ -74,12 +76,40 @@ Use case ends.
...
+Use case ends.
+
+**Extensions**
+
+2a. The list is empty
+
+> Use case ends
+
+3a. The given index is invalid
```
Comment (JoelT-92) [inline doc/DeveloperGuide.md]: Remember that there was no prior mention to an index, furthermore, in this scenario, it should be the system's turn to respond. A good substitute for this line could be:
3a. AddressBook detects that the person selected is invalid.
- 3a1. AddressBook requests for a valid person.
- 3a2. User chooses new person.

Steps 3a1 and 3a2 are repeated until a valid person is chosen.
Use case resumes from step 4.

File: doc/DeveloperGuide.md [hunk 4]
```diff
@@ -74,12 +76,40 @@ Use case ends.
...
+> 3a2. AddressBook shows an error message <br>
+  Use case resumes at step 2
+
 ## Appendix C : Non Functional Requirements
 
 1. Should work on any [mainstream OS](#mainstream-os) as long as it has Java 8 or higher installed.
 2. Should be able to hold up to 1000 persons.
 3. Should come with automated unit tests and open source code.
 4. Should favor DOS style commands over Unix-style commands.
+5. Should be user-friendly enough for non-IT savvy users.
```
Comment (JoelT-92) [inline doc/DeveloperGuide.md]: Not very specific here. You can specify what is user-friendly.

**Pull request closed without merging**